packages/components/src/disclosure/index.tsx

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { DisclosureContentProps } from './types';
+
+/**
+ * Accessible Disclosure component that controls visibility of a section of
+ * content. It follows the WAI-ARIA Disclosure Pattern.
+ *
+ * @see https://ariakit.org/reference/disclosure-content
+ *
+ * The plan is to build own API that accounts for future breaking changes
+ * in Reakit (https://github.com/WordPress/gutenberg/pull/28085).
+ */
+const UnforwardedDisclosureContent = (
+	{ visible, children }: DisclosureContentProps,
+	ref: React.ForwardedRef< any >
+) => {
+	const disclosure = Ariakit.useDisclosureStore( { open: visible } );
```

Comment:
Shouldn't we use a `DisclosureProvider` instead (with the `open` prop), which will allow us to retrieve any needed disclosure context inside the component tree? 

Right now the disclosure store is private to the module and that could also potentially make eventual automated testing more difficult.

Reply:
Retracting this comment as this decision was clarified in https://github.com/WordPress/gutenberg/pull/55639#issuecomment-1794316403.